### DISCUSSION TITLE
How to distinguish ferrite and li'zhuang bainite according to kam

### DISCUSSION THREAD

**Discussion by Mikewangkang** [mtex-toolbox/mtex#1927] [Ask Anything]
hello guys
I need help with a Matlab® code developed that uses the MTEX toolbox to first define grain boundaries based on selected grain tolerance angles and then calculate the KAM for each pixel. If the misorientation of a pixel exceeds 3°, the program classifies the grains as granular bainite; otherwise, it is classified as polygonal ferrite. The classification algorithm then color-codes the relative sorted grains it determines and calculates the measured granular bainite and polygonal fractions. This is my purpose, I have proceeded to calculate the kam value of each pixel but the follow-up process I don't know how to do.
I would be very grateful if anyone could help!
Wang Kang

**Comment by ralfHielscher** [comment]
You can do
1. compute the average kam for each grain `gKAM = grainMean(ebsd,ebsd.KAM,grains)`
2. identify the bainite grains by `grainsB = grains(gKAM > 3*degree)`

I hope this helps,
Ralf.
